app/services/allocation_importer_service.rb

```diff
@@ -0,0 +1,36 @@
+require "csv"
+
+class AllocationImporterService
+  attr_reader :path_to_csv
+
+  def initialize(path_to_csv:)
+    @path_to_csv = path_to_csv
+  end
+
+  def execute
+    rows.each do |row|
+      training_provider = Provider.where(recruitment_cycle_id: 2).find_by(provider_code: row["training_provider_code"])
+      accredited_body_provider = Provider.where(recruitment_cycle_id: 2).find_by(provider_code: row["accredited_body_provider_code"])
```

Comment:
This is probably ok in this case, but generally might be safer not to rely on the recruitment cycle always having id `2`:

```ruby
recruitment_cycle = RecruitmentCycle.current
training_provider = recruitment_cycle.providers.find_by(provider_code: row["training_provider_code"])
accredited_body_provider = recruitment_cycle.providers.find_by(provider_code: row["accredited_body_provider_code"])
```

Reply:
yea i want to be explicit here as we're importing for a specific year. i don't really see it being reused otherwise we could inject the concerning recruitment_cycle. But i don't see that worth doing and see this as throw away code

Reply:
How are we not importing for a specific year? Isn't the data for the 2020-2021 recruitment cycle?

Reply:
> Isn't the data for the 2020-2021 recruitment cycle?

Yes

Sorry, I don't think I follow. `recruitment_cycle_id: 2` is the latest and correct recruitment cycle?

Reply:
It is, I just wasn't clear about the comment being that we're not importing for a specific year. I don't think this is blocking.